libs/s25main/GamePlayer.cpp

```diff
@@ -376,6 +377,19 @@ void GamePlayer::RemoveBuildingSite(noBuildingSite* bldSite)
     buildings.Remove(bldSite);
 }
 
+bool GamePlayer::IsHQTent() const
+{
+    if(const nobHQ* hq = GetHQ())
+        return hq->IsTent();
+    return false;
+}
+
+void GamePlayer::SetHQIsTent(bool isTent)
+{
+    if(nobHQ* hq = const_cast<nobHQ*>(GetHQ()))
```

Comment:
I missed that `GetSpecObj` (in `GetHQ`) can return a non-const pointer. So we don't need the const cast at all.

If you make `GetHQ` non-const then this can be removed